src/main/java/com/faforever/client/mod/ModService.java

```diff
@@ -441,4 +442,30 @@ private void addMod(Path path) {
   public void destroy() {
     Optional.ofNullable(directoryWatcherThread).ifPresent(Thread::interrupt);
   }
+
+  @Async
+  @SneakyThrows
+  public CompletableFuture<List<ModVersion>> updateModVersionsAndActivateThose(final List<ModVersion> selectedModVersions) {
+    final List<ModVersion> newlySelectedMods = new ArrayList<>(selectedModVersions);
+    selectedModVersions.forEach(installedModVersion -> {
+      try {
+        ModVersion modVersionFromApi = getModVersionById(installedModVersion.getUid()).get();
+        ModVersion latestVersion = modVersionFromApi.getMod().getLatestVersion();
+        boolean isLatest = latestVersion.equals(installedModVersion);
```

Comment:
Do these equality actually work since the fields of ModVersion are properties and not the direct values

Reply:
yes it compares uids